### PR TITLE
GW-1262 Refactor handleObject function

### PR DESCRIPTION
**Pull request by smisidjan**

### CODE REVIEW

File: api/src/Service/ObjectEntityService.php [hunk 1]
```diff
@@ -397,151 +357,315 @@ public function handleObject(Handler $handler, Endpoint $endpoint, array $data =
...
+     *
+     * @return ObjectEntity|string[]|void
+     */
+    public function checkGetOperationTypeExceptions(Endpoint $endpoint, Entity $entity, &$data)
+    {
+        if (((isset($operationType) && $operationType === 'item') || $endpoint->getOperationType() === 'item') && array_key_exists('results', $data) && count($data['results']) == 1) { // todo: $data['total'] == 1
+            $data = $data['results'][0];
+            if (isset($data['id']) && Uuid::isValid($data['id'])) {
+                $this->session->set('object', $data['id']);
+            }
```
Comment (rjzondervan) [inline api/src/Service/ObjectEntityService.php:381]: Wouldn't it be better to use the following line:
`isset($data['id']) && Uuid::isValid($data['id']) ?? $this->session->set('object', $data['id']);`
Or something like it? That would rid the function of its nested if.

File: api/src/Service/ObjectEntityService.php [hunk 2]
```diff
@@ -397,151 +357,315 @@ public function handleObject(Handler $handler, Endpoint $endpoint, array $data =
...
+            $fields['_dateRead'] = $fields['_dateRead'] ? 'getItem' : false;
+
+            $data = $this->eavService->handleGet($object, $fields, $extend, $acceptType);
+            if ($object->getHasErrors()) {
+                $data['validationServiceErrors']['Warning'] = 'There are errors, this ObjectEntity might contain corrupted data, you might want to delete it!';
+                $data['validationServiceErrors']['Errors'] = $object->getAllErrors();
+            }
+        } else {
+            $data['error'] = $object;
+        }
```
Comment (rjzondervan) [inline api/src/Service/ObjectEntityService.php:420]: Could we think of something to get rid of the nested ifs without having to rewrite the whole function or break them out to new functions?

Comment (rjzondervan) [reply api/src/Service/ObjectEntityService.php:420]: Hmm..

```
!$object->getSelf() ? $object->setSelf($this->createSelf($object)) : null;
```
and
```
$object->getHasErrors() ? $data['validationServiceErrors'][
  'Warning' => 'There are errors, this ObjectEntity might contain corrupted data, you might want to delete it!',
  'Errors' => $object->getAllErrors(),
] : null;
```

perhaps?

File: api/src/Service/ObjectEntityService.php [hunk 3]
```diff
@@ -397,151 +357,315 @@ public function handleObject(Handler $handler, Endpoint $endpoint, array $data =
...
+            $data = $this->checkGetObjectExceptions($data, $object, $fields, $extend, $acceptType);
+        } else {
+            //todo: -start- old code...
+            //TODO: old code for getting an ObjectEntity
+            $data = $this->eavService->handleSearch($entity, $this->request, $fields, $extend, false, $filters ?? [], $acceptType);
+            //todo: -end- old code...
+
+            if ($this->session->get('endpoint')) {
+                $data = $this->checkGetOperationTypeExceptions($endpoint, $entity, $data);
+            }
```
Comment (rjzondervan) [inline api/src/Service/ObjectEntityService.php:465]: Perhaps you can convert this to a oneliner too:
```
$data = $this->session->get('endpoint') ? $this->checkGetOperationTypeExceptions($endpoint, $entity, $data) : $data;
```

**Review by rjzondervan — COMMENTED** [review]
Good work, seems like a huge improvement.

In the comments there are a few suggestions to mitigate nested if statements, but that's about it.

**Review by WilcoLouwerse — APPROVED** [review]

**Pull request merged**